apollo/apollo.go

```diff
@@ -206,38 +197,45 @@ func (c *client) RegisterConfigCallback(param ConfigParam,
 		callback(data, c.parser)
 	}
 
-	configMap := c.acli.GetNameSpace(param.NameSpace)
+	configMap := c.acli.GetNameSpace(param.nameSpace)
 	data, ok := configMap[param.Key]
 	if !ok {
 		klog.Info("key:", param.Key)
 		klog.Info("CONFIG:", configMap)
-		panic(errors.New("kitex config : key not found"))
+		klog.Debug("[apollo] key not found")
+	} else {
+		callback(data.(string), c.parser)
 	}
-	// data := c.acli.Get(param.Key, agollo.WithNamespace(param.NameSpace))
-	callback(data.(string), c.parser)
 
 	go c.listenConfig(param, onChange)
 }
 
 func (c *client) listenConfig(param ConfigParam, callback func(namespace, cluster, key, data string)) {
+	defer func() {
+		if err := recover(); err != nil {
+			klog.Error("[apollo] listen goroutine error:", err)
+		}
+	}()
 	errorsCh := c.acli.Start()
-	apolloRespCh := c.acli.WatchNamespace(param.NameSpace, make(chan bool))
+	apolloRespCh := c.acli.WatchNamespace(param.nameSpace, make(chan bool))
```

Comment:
I'm a bit confused about the usage of the apollo client,  it calls the `Start` function when listening to a new config and it can be called multiple times. The apollo SDK uses `sync.Once` to avoid the multiple calls, but we should not rely on this mechanism, the same about the `Stop`.

Reply:
Your mean is that the `Start` function shoud be placed in a global Configuration？

Reply:
Yes, and it seems that the Agollo client is not stopped, even though it receives the error msg from the channel.